lib/article_json/export/common/html/elements/image.rb

```diff
@@ -18,13 +26,18 @@ def export
               end
             end
 
-            private
-
             # @return [Nokogiri::XML::NodeSet]
             def image_node
               create_element(:img, src: @element.source_url)
             end
 
+            # @return [Nokogiri::XML::NodeSet]
+            def href_node
+              create_element(:a, href: @element.href) do |a|
+                a.add_child(figure_node)
```

Comment:
What would currently happen if the figure had a caption? Would the caption also be wrapped in the `<a>` tag? I am mainly asking because there may be other links in the caption, which would then mean nested `<a>` tags (and I have no clue how different browsers would handle that..). Therefore, I'd recommend only nesting the `<img>` tag within the `<a>` and keeping the caption outside.

It'd also be good to have a test for this to see what happens if an image has both an `href` and a `caption`.

Reply:
Good point!

Reply:
@nicolas-fricke Changed!

Reply:
Great, thanks!